models/repo_indexer.go

```diff
@@ -70,34 +70,58 @@ func InitRepoIndexer() {
 	if !setting.Indexer.RepoIndexerEnabled {
 		return
 	}
-	indexer.InitRepoIndexer(populateRepoIndexer)
+	indexer.InitRepoIndexer(populateRepoIndexerAsynchronously)
 	repoIndexerOperationQueue = make(chan repoIndexerOperation, setting.Indexer.UpdateQueueLength)
```

Comment:
race condition: repoIndexerOperationQueue is initialized after indexer.InitRepoIndexer is called. While I agree it is unlikely that L117 is reached before L74, better be safe than sorry.

Reply:
Good catch, fixed